.taskcluster.yml

```diff
@@ -1,160 +1,127 @@
-      artifacts:
-        'public':
-          type: 'directory'
-          path: '/opt/firefox-tv/app/build/reports'
-          expires: "{{ '1 week' | $fromNow }}"
-
-      features:
-          taskclusterProxy: true
-
-    metadata:
-      name: Firefox for Amazon's Fire TV - Build - Master
-      description: Building Firefox for Amazon's Fire TV (via Gradle) - Master
-      owner: '{{ event.head.user.email }}'
-      source: '{{ event.head.repo.url }}'
-
-###############################################################################
-# Task: Release builds
-#
-# Triggered when a new tag or release is published (in any branch)
-#
-# - Build (unsigned) release versions of the app with release translations and
-#   secrets.
-###############################################################################
-  - provisionerId: '{{ taskcluster.docker.provisionerId }}'
-    workerType: '{{ taskcluster.docker.workerType }}'
-    extra:
-      github:
-        events:
-          - release
-    scopes:
-      - "secrets:get:project/mobile/firefox-tv/tokens"
-    payload:
-      maxRunTime: 3600
-      deadline: "{{ '2 hours' | $fromNow }}"
-      image: 'mozillamobile/firefox-tv:2.3'
-      command:
-        - /bin/bash
-        - '--login'
-        - '-cx'
-        - >-
-          git fetch origin --tags
-          && git config advice.detachedHead false
-          && git checkout {{ event.version }}
-          && yes | sdkmanager --licenses
-          && python tools/taskcluster/get-sentry-token.py
-          && python tools/taskcluster/get-pocket-token.py
-          && ./gradlew --no-daemon clean test assembleSystemRelease
-      artifacts:
-        'public':
-          type: 'directory'
-          path: '/opt/firefox-tv/app/build/outputs/apk'
-          expires: "{{ '1 week' | $fromNow }}"
-      features:
-        taskclusterProxy: true
-    metadata:
-      name: Firefox for Fire TV - Release build ({{ event.version }})
-      description: Building Firefox for Amazon's Fire TV (via Gradle) - Release
-      owner: '{{ event.head.user.email }}'
-      source: '{{ event.head.repo.url }}'
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+    trust_level:
+      $if: 'tasks_for in ["github-push", "github-release"] && event.repository.html_url == "https://github.com/mozilla-mobile/firefox-tv"'
+      # Trust Level number corresponds to the "trust level" constants from the main Firefox repository.
+      # 3 represents trusted work (it's been reviewed and landed, or it's work by a trusted dev in-repo) while
```

Comment:
Do you recommend core devs work in repo then? What does this trust level affect?

Reply:
Ah, good question! Core devs should still follow the same style (do work in separate branch, PR it in).

This trust level affects signing and pushing apks - we shouldn't use our signing key or amazon credentials unless we can "trust" the commit (and we know it can be trusted if it's part of this repo).